txpipe/source_selector.py

```diff
@@ -441,18 +443,36 @@ def write_global_values(self, outfile, calibrators, number_density_stats):
         C = np.zeros((nbin_source,1,2))
         N = np.zeros(nbin_source)
         R_scalar = np.zeros(nbin_source)
+        mean_e1 = np.zeros(nbin_source)
+        mean_e2 = np.zeros(nbin_source)
+
+        means, sigma_e = number_density_stats.collect()
 
-        # this needs fixing
-        sigma_e = number_density_stats.collect()
 
         for i, cal in enumerate(calibrators):
+            mu1 = np.array([means[i, 0]])
+            mu2 = np.array([means[i, 1]])
             if self.config['shear_catalog_type']=='metacal':
+                # Collect the total calibration factor
                 R[i], S[i], N[i] = cal.collect(self.comm)
-                sigma_e[i] /= 0.5*(R[i,0,0] + R[i,1,1])
-            else:
+
+                # Apply it to the means
+                mean_e1[i], mean_e2[i] = apply_metacal_response(
+                    R[i], S[i], g1=mu1, g2=mu2)
+
+                # TODO: use full R matrix here?
+                sigma_e[i] /= 0.5 * (R[i,0,0] + R[i,1,1])
```

Comment:
Just had a think about this - I think we'd need to calculate `<e_1 * e_2`> to do it in full, since that's what we'd apply the off-diaonal R matrix elements to.  The difference should be small, though, since the off-diagonals are small, and this is only on sigma_e, so we don't need the same high acc that we need on 

Having said that, this current correction can be improved - I'll push something.

Reply:
Have pushed a change to this.